api/v1/client/client.go

```diff
@@ -29,29 +29,81 @@ import (
 	"net/http"
 	"net/url"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/manyminds/api2go/jsonapi"
 
-	"github.com/loadimpact/k6/api/v1"
+	v1 "github.com/loadimpact/k6/api/v1"
 )
 
+// Client is a simple HTTP client for the REST API.
 type Client struct {
-	BaseURL *url.URL
+	BaseURL    *url.URL
+	httpClient *http.Client
+	logger     *logrus.Entry
 }
 
-func New(base string) (*Client, error) {
+// Option function are helpers that enable the flexible configuration of the
+// REST API client.
+type Option func(*Client)
+
+// New returns a newly configured REST API Client.
+func New(base string, options ...Option) (*Client, error) {
 	baseURL, err := url.Parse("http://" + base)
 	if err != nil {
 		return nil, err
 	}
-	return &Client{BaseURL: baseURL}, nil
+	c := &Client{
+		BaseURL:    baseURL,
+		httpClient: http.DefaultClient,
+	}
+
+	for _, option := range options {
+		option(c)
+	}
+
+	return c, nil
+}
+
+// WithHTTPClient configures the supplied HTTP client to be used when making
+// REST API requests.
+func WithHTTPClient(httpClient *http.Client) Option {
+	return Option(func(c *Client) {
+		c.httpClient = httpClient
+	})
+}
+
+// WithLogger sets the specifield logger to the client.
+func WithLogger(logger *logrus.Entry) Option {
+	return Option(func(c *Client) {
+		c.logger = logger
+	})
 }
 
-func (c *Client) call(ctx context.Context, method string, rel *url.URL, body, out interface{}) error {
+// Call executes the desired REST API request.
+func (c *Client) Call(ctx context.Context, method string, rel *url.URL, body, out interface{}) (err error) {
```

Comment:
This is now exported because it's called from... where again? :)

Reply:
Places